feat: toggle for advanced operating tables preventing rot

perryprog opened the pull request:
<!-- If you are new to the Delta-V repository, please read the [Contributing Guidelines](https://github.com/DeltaV-Station/Delta-v/blob/master/CONTRIBUTING.md) -->

## About the PR
<!-- What did you change? -->
Adds a toggle to advanced surgery tables to switch between surgery speed boost and antirot.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
If you're working on a dead patient that has eight million burn then you should at least be allowed to use an autodoc to help.

## Technical details
<!-- Summary of code changes for easier review. -->

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc).
Small fixes/refactors are exempt. -->
![image](https://github.com/user-attachments/assets/cc054280-5a6d-44af-9878-6bf5c14d45f5)
![image](https://github.com/user-attachments/assets/39cf9207-d05d-4ae4-96bd-13e0b8469c22)


## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have tested all added content and changes.
- [X] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelogs must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->
:cl:
- add: Advanced operating tables can now be switched between preventing rot and increasing surgery speed.

perryprog committed:
feat: allow using autodocs with stasis beds

Stop-Signs commented:
Direction here, we'd like this to be a trait that the advanced operating table has in order to give players a better reason to make it.

perryprog commented:
Sure—to be clear, are you saying the advanced operating table should pause (or slow down?) rot?

Stop-Signs commented:
> Sure—to be clear, are you saying the advanced operating table should pause (or slow down?) rot?

Ok so, I did actually misread something in our thread. This should either be a *mode* on the advanced operating table. or it should be its own thing, like an advanced stasis bed.

apologies for the mistake there, thank you for asking.

perryprog commented:
Sure. And what would the mode on the advanced operating table look like? As in, in what way does it change its behavior?

Stop-Signs commented:
> Sure. And what would the mode on the advanced operating table look like? As in, in what way does it change its behavior?

Something where you can toggle between faster surgery speed and stasis.

perryprog committed:
feat: let advanced operating tables work as stasis beds instead

perryprog commented:
Okie, here's what I got; see updated PR description and media.

Not sold on my wording choices and my verbs are in fact not verbs, but I can't think of anything better.

perryprog committed:
Slightly better operating table verbs

perryprog committed:
Really weird test flake

deltanedas commented on Content.Shared/_DV/Item/ItemToggle/Components/ComponentTogglerComponent.cs:
```diff
@@ -0,0 +1,18 @@
+    /// </summary>
+    [DataField]
+    public ComponentRegistry? ComponentsAddedOnDeactivate;
+}
```
dont need 2 of the same thing, rename to RemovedComponents and use it for both cases
then remove the field-less one from the yml

perryprog replied:
So true. Since upstream has 'RemoveComponents' (only used on deactivate, so not helpful here) I went with 'DeactivatedComponents'.

deltanedas commented on Resources/Prototypes/_Shitmed/Entities/Structures/Machines/advanced_operating_table.yml:
```diff
@@ -8,3 +8,20 @@
+    - type: SurgerySpeedModifier
+    componentsAddedOnDeactivate:
+    - type: SurgerySpeedModifier
+      speedModifier: 1.25
```
missing end comment

Stop-Signs reviewed (approved):
Direction approved, this is fantastic work.

perryprog committed:
refactor: address review

- Missing end DeltaV comment
- Merge new ToggleComponent fields to DeactivatedComponents

deltanedas commented on Resources/Locale/en-US/_DV/surgery/advanced-operating-table.ftl:
```diff
@@ -0,0 +1,5 @@
+advanced-operating-table-stasis-mode = It is currently [color=lightblue]slowing down decay[/color] of patients strapped to it.
+advanced-operating-table-surgery-mode = It is currently [color=lightgreen]stabilizing the patient's movement[/color].
```
reword it, currently it implies it wouldnt make surgery faster for a corpse since it doesnt move
probably just say speeds up surgery its not a crazy secret

perryprog replied:
Yeah, good point on both counts.

SirWarock commented:
Will the advanced operating table still work like a medical bed?
Aka, it heals blunt and poison damage, which is a really neat feature for when you forget to clean your tools, but the patient auto heals the small amount you caused.

perryprog committed:
Tweak advanced operating table wording

perryprog commented:
> Will the advanced operating table still work like a medical bed? Aka, it heals blunt and poison damage, which is a really neat feature for when you forget to clean your tools, but the patient auto heals the small amount you caused.

Yeah, it only looses the surgery speed enhancement when in "stasis bed mode", and stasis bed mode only prevents rot, nothing else. Worth noting that it's getting that because it is inheriting from normal beds, not medical beds—medical beds passively heal more.

Also bed passive healing doesn't work on dead anyway.
